### ISSUE TITLE
Big delay when using bluetooth microphone (airpods)

### ISSUE THREAD

**Issue by alexgurr** [Streampunk/naudiodon#52]
I'm using this library to record the user's audio input stream. With a built-in microphone it's perfect. However when I use my airpods microphone, there's such an initial buffer/delay that the first 1/2 seconds of talking are cut off. Any ideas on how to improve this?

Code for reference:

```
  audioRecorder.ai = new portAudio.AudioIO({
    inOptions: {
      channelCount: inputDevice.maxInputChannels,
      sampleFormat: portAudio.SampleFormat16Bit,
      sampleRate: 44100,
      deviceId: inputDevice.id,
      closeOnError: false 
    }
  });

  audioRecorder.channels = inputDevice.maxInputChannels;

  // Create a write stream to write out to a raw audio file
  const ws = require('fs').createWriteStream(`tmp/${title}.raw`)

  //Start streaming
  audioRecorder.ai.pipe(ws);
  audioRecorder.ai.start();
```

**Comment by scriptorian** [comment]
I don't understand what's going on but I can offer two more inOptions to experiment with to see if it makes any difference:
 - _highwaterMark_: the JS stream parameter, defaults to 16384 bytes
 - _maxQueue_: queue of internal buffers that receive data from the device, defaults to 2

Hopefully something of this will help - or at least make a difference.

**Comment by alexgurr** [comment]
Hi Simon, thanks for reaching out.

The code I posted above is pretty much the same as the default example in the readme. Which bits don't you get?

I will take a look at these parameters - what's the source for them? Do you have any links to docos?

**Comment by alectrocute** [comment]
@alexgurr I don't think this issue is specific to `naudiodon`, but rather a lower-level issue with `coreaudio`. The same issue, when using AirPod 1/2/Pro occurs in Chromium with the `getUserMedia` API. First second of audio is cut off even though the stream is active.
